mailpoet/lib/Automation/Integrations/MailPoet/Analytics/Controller/AutomationEmailController.php

```diff
@@ -0,0 +1,93 @@
+<?php declare(strict_types = 1);
+
+namespace MailPoet\Automation\Integrations\MailPoet\Analytics\Controller;
+
+use MailPoet\Automation\Engine\Data\Automation;
+use MailPoet\Automation\Engine\Storage\AutomationStorage;
+use MailPoet\Automation\Integrations\MailPoet\Actions\SendEmailAction;
+use MailPoet\Entities\NewsletterEntity;
+use MailPoet\Newsletter\NewslettersRepository;
+
+class AutomationEmailController {
+
+
+  /** @var AutomationStorage */
+  private $automationStorage;
+
+  /** @var NewslettersRepository */
+  private $newslettersRepository;
+
+  public function __construct(
+    AutomationStorage $automationStorage,
+    NewslettersRepository $newslettersRepository
+  ) {
+    $this->automationStorage = $automationStorage;
+    $this->newslettersRepository = $newslettersRepository;
+  }
+
+  /**
+   * @param Automation $automation
+   * @param \DateTimeImmutable $after
+   * @param \DateTimeImmutable $before
+   * @return NewsletterEntity[]
+   */
+  public function getAutomationEmailsInTimeSpan(Automation $automation, \DateTimeImmutable $after, \DateTimeImmutable $before): array {
+    $automationVersions = $this->automationStorage->getAutomationVersionDates($automation->getId());
+    usort(
+      $automationVersions,
+      function (array $a, array $b) {
+        return $a['created_at'] <=> $b['created_at'];
+      }
+    );
+
+    // filter automations that were created before the after date
+    $versionIds = [];
+    foreach ($automationVersions as $automationVersion) {
+      if ($automationVersion['created_at'] > $before) {
+        break;
+      }
+      if (!$versionIds || $automationVersion['created_at'] < $after) {
+        $versionIds = [(int)$automationVersion['id']];
+        continue;
+      }
+      $versionIds[] = (int)$automationVersion['id'];
+    }
+
+    $automations = $this->automationStorage->getAutomationWithDifferentVersions($versionIds);
```

Comment:
This will become its own function and the name of this class will be renamed to `AutomationTimeSpanController` in the Automation Flow Tab PR.

Not super happy with the name and location of the class.